src/odemis/driver/xt_client.py

```diff
@@ -1616,6 +1622,65 @@ def _get_acq_msg(self, **kwargs):
         return msg
 
 
+class Chamber(model.Actuator):
+    """
+    Component representing the vaccum chamber. Changing the pressure is possible by moving the "vacuum" axis,
+    which accepts two position values: 0 for vented, 1 for vacuum.
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        axes = {"vacuum": model.Axis(choices={0: "vented", 1: "vacuum"})}
+        model.Actuator.__init__(self, name, role, parent=parent, axes=axes, **kwargs)
+
+        self.position = model.VigilantAttribute({"vacuum": 0}, readonly=True)
+        self.pressure = model.FloatContinuous(self.parent.get_pressure(), range=(0, 1e9), readonly=True, unit="Pa")
+        self._polling_thread = util.RepeatingTimer(5, self._refreshPressure, "Pressure polling")
+        self._polling_thread.start()
+
+        self._executor = CancellableThreadPoolExecutor(max_workers=1)
+
+    def stop(self, axes=None):
+        raise NotImplementedError("Stopping the pumping/venting process is not supported.")
```

Comment:
Please don't that, because it might cause some failures when we have an emergency stop (which stops *all* axes).
You can do a `self._executor.cancel()`, and even log a warning if you detect that there is a move in progress (to be fancy).

Reply:
How would I detect that a move is in progress?

Reply:
There is no super official way. However, you can check `if self._executor._queue`.